Feature/tabs bar composable

MuhammedEdrees opened the pull request:
Implemented Tabs component from design system.
Attached is a demo of the TabsBar in action.

https://github.com/user-attachments/assets/68e73145-0174-4af7-9f55-3a85fdadddae

ABDULLAHHG committed:
add nunito font styles to resource

ABDULLAHHG committed:
add a data class for tudee text style

Arzo-zi committed:
add tudeeColor to handle light and dark theme

Arzo-zi committed:
refactor theme management for light and dark

ABDULLAHHG committed:
implement tudee text style font

ABDULLAHHG committed:
Merge branch 'feature/color-design-lightmode' into feature/design-system-font-declear

ABDULLAHHG committed:
Refactor textStyle package location and integrate typography into theme

Hasa9nAli committed:
call content of application inside TudeeTheme tree

Hasa9nAli committed:
add all drawable icons of project

baraa0abd committed:
Add deployment target selector configuration and robot images

This commit introduces the following:
- Adds `.idea/deploymentTargetSelector.xml` file to configure deployment target selection for the app run configuration.
- Adds various density versions (mdpi, hdpi, xhdpi, xxhdpi, xxxhdpi) of three different robot images:
    - `749d09de62af3aa8ec816c102a9dce71959de02a.png` (happy robot)
    - `Frame 2095586230.png` (frame)
    - `Icon.png` (icon)
- Organizes these images into `robotimages`, `robotimage2`, and `robotimage3` subdirectories within `app/src/main/res/drawable/`.
- Also adds these images directly into density-specific drawable folders (e.g., `app/src/main/res/drawable-hdpi/`) with descriptive names like `img_happy_robot.png`, `img_frame_hdpi.png`, and `img_icon_hdpi.png`.

Hasa9nAli committed:
add SnakeBar composable for displaying messages with status indicators

Hasa9nAli committed:
Merge remote-tracking branch 'origin/feature/design-system' into feature/design-system

Hasa9nAli committed:
Add DayChip composable for displaying selectable day options

Hasa9nAli committed:
Add Priority composable for displaying task priority indicators

MuhammedEdrees committed:
Ignore Android Studio's .idea directory

MuhammedEdrees committed:
implement TabsBar composable

MuhammedEdrees committed:
bump compileSdk and targetSdk to 35

MuhammedEdrees committed:
refactor TabsBar to improve readability

MahmoodTarek reviewed (approved):
nice work Edress , it's very helpful code

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+    selectedTabIndex: Int,
+    onTabClicked: (index: Int) -> Unit,
+    modifier: Modifier = Modifier
+) {
```
I believe the naming should not be plural, so this function name should be something as TabBar, and the property tabs be tab

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+fun TabsBar(
+    tabs: List<TabItem>,
+    selectedTabIndex: Int,
+    onTabClicked: (index: Int) -> Unit,
```
I believe the naming should not be plural, so this function name should be something as TabBar, and the property tabs be tab

MuhammedEdrees replied:
Done! Thanks for the review

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+    onTabClicked: (index: Int) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val borderColor = Theme.color.textColor.stroke
```
pass it directly into borderColor, no need to store it into variable

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+@Composable
+fun TabsBar(
+    tabs: List<TabItem>,
+    selectedTabIndex: Int,
```
I think we can improve the functionality here, you can create a data class, for example called 'Selectable'
and it should be generic, it has 2 properties (value:T , isSelected:Boolean = false)
and pass tabs here like this List<Selectable<TabItem>>
also rename the action to 'onTabSelected' and has the TabItem passed as parameter

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+            Tab(
+                modifier = Modifier.weight(1f),
+                tab = tab,
+                isSelected = selectedTabIndex == index,
```
then isSelected here would be 'isSelected = tab.isSelected' Isn't that better!

MuhammedEdrees replied:
much better! will change it

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val secondaryColor = Theme.color.secondaryColor
```
No need to store colors like that, please pass them directly

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabsBar.kt:
```diff
@@ -0,0 +1,192 @@
+    color: Color,
+    underlinePath: Path
+) = this.wrapContentWidth().drawBehind {
+    if (isSelected) {
```
use AnimatedVisibility instead

ahmednasserzaza reviewed (changes requested):
Well done
Check the requested changes please

Hasa9nAli reviewed (commented):
Keep going Muhammed

MuhammedEdrees committed:
add Selectable data class

MuhammedEdrees committed:
rename TabsBar to TabBar and refactor for Selectable tabs

MuhammedEdrees commented:
Last push added the following requested changes:
- Added Selectable generic class and utilized it.
- Added animation for the underline in selected tab.
- Various improvements for performance and code readability.

Attached is a demo of the tab bar in action. 

https://github.com/user-attachments/assets/39c7e305-d3b1-451d-9aa4-29569f3c4a34

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/TabBar.kt:
```diff
@@ -0,0 +1,223 @@
+
+@Composable
+fun TabBar(
+    selectableTabs: List<Selectable<TabItem>>,
```
just tabs

MuhammedEdrees committed:
rename TabBar to Tabs

MuhammedEdrees committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feature/tabs-bar-composable

# Conflicts:
#	.idea/deploymentTargetSelector.xml
#	app/src/main/java/com/baghdad/tudee/designSystem/color/TudeeColor.kt
#	app/src/main/java/com/baghdad/tudee/designSystem/textStyle/TudeeTextStyle.kt
#	app/src/main/java/com/baghdad/tudee/designSystem/textStyle/defaultTextStyle.kt
#	app/src/main/java/com/baghdad/tudee/designSystem/theme/TudeeTheme.kt
#	app/src/main/java/com/baghdad/tudee/ui/composable/DayChip.kt
#	app/src/main/java/com/baghdad/tudee/ui/composable/SnakeBar.kt
#	app/src/main/res/drawable/ic_thinking_person.xml

hagarabobakr commented on app/src/main/java/com/baghdad/tudee/ui/composable/Tabs.kt:
```diff
@@ -0,0 +1,223 @@
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Center
+        ) {
+            BasicText(
```
It is better to replace a BasicText with Text to maintain consistency

hagarabobakr commented on app/src/main/java/com/baghdad/tudee/ui/composable/Tabs.kt:
```diff
@@ -0,0 +1,223 @@
+            ),
+        contentAlignment = Alignment.Center
+    ) {
+        BasicText(
```
Here too

Hasa9nAli replied:
hi hager thank you for your effort , muhammed here use `BasicText` to avoid use any thing related on material 3

hagarabobakr commented:
Great job 👏
